From 647daad876ec0511442998906e7826541ec6e459 Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Thu, 9 Jul 2026 15:17:17 +0100
Subject: [PATCH] Disable per-field profile privacy - always public

Remove the privacy picker, and make all profile fields always visible to all viewers.

---
 UserProfile/includes/SPUserSecurity.php | 100 +-----------------------
 1 file changed, 2 insertions(+), 98 deletions(-)

diff --git a/UserProfile/includes/SPUserSecurity.php b/UserProfile/includes/SPUserSecurity.php
index 78783548..dbc2e978 100644
--- a/UserProfile/includes/SPUserSecurity.php
+++ b/UserProfile/includes/SPUserSecurity.php
@@ -79,55 +79,7 @@ class SPUserSecurity {
 	 * @return string HTML suitable for output
 	 */
 	public static function renderEye( $fieldKey, User $user ) {
-		$dbw = MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnection( DB_PRIMARY );
-		$s = $dbw->selectRow(
-			'user_fields_privacy',
-			[ '*' ],
-			[ 'ufp_field_key' => $fieldKey, 'ufp_actor' => $user->getActorId() ],
-			__METHOD__
-		);
-
-		if ( $s && !empty( $s->ufp_privacy ) ) {
-			$privacy = $s->ufp_privacy;
-		} else {
-			$privacy = 'public';
-		}
-
-		// Form list with remaining privacies
-		$all_privacy = [ 'public', 'hidden', 'friends', 'foaf' ];
-
-		$ret = '<div class="eye-container" current_action="' .
-			htmlspecialchars( $privacy, ENT_QUOTES ) . '" fieldkey="' .
-			htmlspecialchars( $fieldKey, ENT_QUOTES ) . '">
-					<div class="title">' .
-					// For grep: i18n messages used here:
-					// user-profile-privacy-status-privacy-public,
-					// user-profile-privacy-status-privacy-hidden,
-					// user-profile-privacy-status-privacy-friends,
-					// user-profile-privacy-status-privacy-foaf
-					wfMessage( 'user-profile-privacy-status-privacy-' . $privacy )->escaped() . '</div>
-					<div class="menu">';
-		$noscriptVersion = '<noscript><select name="' . htmlspecialchars( $fieldKey, ENT_QUOTES ) . '">';
-
-		foreach ( $all_privacy as $priv ) {
-			if ( $priv == $privacy ) {
-				$noscriptVersion .= '<option value="' . htmlspecialchars( $privacy, ENT_QUOTES ) .
-					'" selected="selected">' . wfMessage( 'user-profile-privacy-status-privacy-' . $privacy )->escaped() . '</option>';
-				continue;
-			}
-
-			$ret .= '<div class="item" action="' . htmlspecialchars( $priv, ENT_QUOTES ) . '">' .
-				wfMessage( 'user-profile-privacy-status-privacy-' . $priv )->escaped() .
-				'</div>';
-			$noscriptVersion .= '<option value="' . htmlspecialchars( $priv, ENT_QUOTES ) . '">' .
-				wfMessage( 'user-profile-privacy-status-privacy-' . $priv )->escaped() . '</option>';
-		}
-
-		$ret .= '</div>
-			</div>';
-		$noscriptVersion .= '</select></noscript>';
-
-		return $ret . $noscriptVersion;
+		return '';
 	}
 
 	/**
@@ -172,55 +124,7 @@ class SPUserSecurity {
 	 * @return bool True if the user can view the field, otherwise false
 	 */
 	public static function isFieldVisible( $owner, $fieldKey, $viewer = null ) {
-		// No viewing user supplied -> use the current user
-		if ( $viewer == null ) {
-			$viewer = RequestContext::getMain()->getUser();
-		}
-
-		// Owner can always view all of their profile fields, obviously
-		if ( $viewer->getActorId() == $owner->getActorId() ) {
-			return true;
-		}
-
-		$relation = UserRelationship::getUserRelationshipByID( $viewer, $owner ); // 1 = friend, 2 = foe
-		$privacy = self::getPrivacy( $owner, $fieldKey );
-
-		switch ( $privacy ) {
-			case 'public':
-				return true;
-
-			case 'hidden':
-				return false;
-
-			case 'friends':
-				if ( $relation == 1 ) {
-					return true;
-				}
-				break;
-
-			case 'foaf':
-				if ( $relation == 1 ) {
-					return true;
-				}
-
-				// Now we know that the viewer is not the user's friend, but we
-				// must check if the viewer has friends that are the owner's friends:
-				$listLookup = new RelationshipListLookup( $owner );
-				$ownerFriends = $listLookup->getFriendList();
-
-				foreach ( $ownerFriends as $friend ) {
-					// If someone in the owner's friends has the viewer in their
-					// friends, the test is passed
-					$friendActor = User::newFromActorId( $friend['actor'] );
-					if ( UserRelationship::getUserRelationshipByID( $friendActor, $viewer ) == 1 ) {
-						return true;
-					}
-				}
-
-				break;
-		}
-
-		return false;
+		return true;
 	}
 
 }
